docs/building-rpm-ostree-and-autosd.md

```diff
@@ -0,0 +1,171 @@
+RUN cd rpmbuild/SPECS/ && curl -OL https://gitlab.com/redhat/centos-stream/rpms/rpm-ostree/-/raw/c9s/rpm-ostree.spec
+RUN cd /tmp && dnf builddep -y rpm-ostree.spec
+
+# Lets use the latest from community (see next curl rustup step).
+# Also avoids "cargo metadata: error: failed to run rustc to learn about target-specific information"
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+```
+
+## Building the image
+
+```bash
+podman build -t rpmostreedevel -f ContainerFile
+```
+
+## Running the image
+
+The strategy now is run the container sharing the current **HOME/rpmbuild** dir, so as soon the rpm is created in the container it will be located in the host machine in the running user dir.
+
+```bash
+sudo podman run --rm --privileged -v /home/$USER/rpmbuild/:/home/$USER/rpmbuild/ -ti rpmostreedevel /bin/bash
+
+[root@d1ec4ed543ec /]#
+```
+
+## Building a forked rpm-ostree
+
+Now time to clone your [rpm-ostree](https://github.com/coreos/rpm-ostree) fork and build it.
+
+Cloning the repo and switch to your work branch
+
+```bash
+cd /tmp
+git clone https://github.com/dougsland/rpm-ostree.git && cd rpm-ostree
+git checkout remotes/origin/initoverlayfs -b initoverlayfs
+```
+
+Adding required submodules for the build
+
+```bash
+git submodule update --init
+```
+
+Generating source code
+
+```bash
+cd packaging
+cargo install cargo-vendor-filterer --version ^0.5
+make -f Makefile.dist-packaging
+cp rpm-ostree-e2dbedd3.tar.xz $HOME/rpmbuild/SOURCES/
+
+# Getting version and updating the spec file
+rpmostree_tarxz=$(ls rpm-ostree-*.tar.xz| sed 's/rpm-ostree-\(.*\)\.tar\.xz/\1/')
+sed -i "/^Version:/s/:.*/: $rpmostree_tarxz/" $HOME/rpmbuild/SPECS/rpm-ostree.spec
+
+# Remove any specific Patch for Fedora koji build
+# See:
+# https://gitlab.com/redhat/centos-stream/rpms/rpm-ostree/-/blob/8e861dccda62f170461da2b90cc2f549886094aa/0001-cliwrap-rpm-mark-eval-E-as-safe.patch#L14
+# https://gitlab.com/redhat/centos-stream/rpms/rpm-ostree/-/blob/8e861dccda62f170461da2b90cc2f549886094aa/0001-cliwrap-rpm-mark-eval-E-as-safe.patch
+sed -i '/Patch[0-9]*:/d' $HOME/rpmbuild/SPECS/rpm-ostree.spec
+```
+
+Keep in mind, the version of rpm-ostree must be higher than
+the current version shipped in the autosd image. In the time this document was created the source file generated in the rpm-ostree is based on hash commit which is not enough most of the times. Below a quick hack.
+
+```bash
+# untar the source file:
+tar xvf rpm-ostree-e2dbedd3.tar.xz
+
+# rename the dir
+mv rpm-ostree-e2dbedd3 rpm-ostree-2025.11
+
+# tar again
+tar -cJf rpm-ostree-2025.11.tar.xz rpm-ostree-2025.11
+
+# change the spec file to use this new version, replace in the
+# spec file the Version field to Version: 2025.11
+vi /root/rpmbuild/SPECS/rpm-ostree.spec
+```
+
+Finally generate the RPM
+
+```bash
+rpmbuild -ba ~/rpmbuild/SPECS/rpm-ostree.spec
+```
+
+You are done, exit from the container
+
+```bash
+exit
+```
+
+## Building autosd with the generated file
```

Comment:
It seems to me like everything past here could link to a doc that lives in autosd's repository like "here's how you use a locally built RPM", right?

Reply:
Yes but would make sense to at least link both urls if merged in different github projects.